### DISCUSSION TITLE
[Feedback forum] SCIM for GHES 3.6+ private beta

### DISCUSSION THREAD

**Discussion by hpsin** [community/community#36825] [Enterprise]
**Current beta status**: Development for this private beta has been paused temporarily. Current beta customers may continue to test but are strongly advised to refrain from production use as the beta is currently unsupported.

In GitHub Enterprise Server 3.6+, a [new SCIM API](https://docs.github.com/en/enterprise-server@latest/rest/enterprise-admin/scim) was included for testing purposes. This API is disabled by default, as it is not yet supported.  If you are interested in testing this API, please work with your account team at GitHub to enroll in the private beta.  While we are primarily testing against the Azure AD and Okta SCIM dialects, we are providing the API for open standards usage - you may use any mix of SAML and SCIM providers to authenticate and provision GHES. 

The SCIM API ([REST API documentation](https://docs.github.com/en/enterprise-server@latest/rest/enterprise-admin/scim)) allows a SCIM provider to provision accounts into your GHES instance, rather than using the "just-in-time" (JIT) provisioning that SAML authentication provides.  SCIM also supports the syncing of assigned groups, which are used to sync teams in GitHub with groups in your IdP. You can learn more about this feature in the [Enterprise Managed User documentation](https://docs.github.com/en/enterprise-cloud@latest/admin/identity-and-access-management/using-enterprise-managed-users-for-iam/managing-team-memberships-with-identity-provider-groups). 

This SCIM API operates like the Enterprise Managed User SCIM provisioning model, with two caveats:
1. Migration from SAML JIT to SCIM provisioning is a new flow, where existing GHES users are matched to incoming SCIM provisioning events using their username.
2. Support for ["mixed mode" authentication](https://docs.github.com/en/enterprise-server@latest/admin/identity-and-access-management/managing-iam-for-your-enterprise/allowing-built-in-authentication-for-users-outside-your-provider), where only _some_ users are SCIM provisioned and use SAML for authentication, while others use built-in accounts to authenticate.  

If you're interested in the private beta, to help GitHub test this new API with a variety of SCIM and SAML providers and ensure that your IdP supports this authentication pattern, please reach out to your account team. Please note that this private beta is unsupported - support tickets for the SCIM API in 3.6+ will be redirected to your account team and the SCIM API engineering team for help and feedback. Thank you for your help!

**Comment by moma8468** [comment]
In light of release 3.8, 3.9 and 3.10, could you please update the status?

**Reply by gpadak** [reply]
Support has been resumed for this private beta as of 3.11.9, 3.12.3, and 3.13 ongoing: https://docs.github.com/en/enterprise-server@3.11/admin/release-notes#3.11.9-changes

**Comment by gpadak** [comment]
SCIM support for GHES is now in public beta in 3.14. Please head to this discussion thread with any feedback on the 🚢 ❤️ https://github.com/orgs/community/discussions/136986
